client/src/redux/shopCart/cart-reducer.js

```diff
@@ -0,0 +1,44 @@
+import CartActionTypes from "./cart-types";
+import { addItemToCart, removeItemFromCart } from "./cart-utils";
+
+const INITIAL_STATE = {
+  hidden: true,
+  cartItems: []
+};
+
+const cartReducer = (state = INITIAL_STATE, action) => {
+  switch (action.type) {
+    case CartActionTypes.TOGGLE_CART_HIDDEN:
+      return {
+        ...state,
+        hidden: !state.hidden
+      };
+    case CartActionTypes.ADDED_ITEM:
+    case CartActionTypes.INCREMENTED_ITEM_QUANTITY:
+      return {
+        ...state,
+        cartItems: addItemToCart(state.cartItems, action.payload)
+      };
+    case CartActionTypes.REMOVED_ITEM:
```

Comment:
`REMOVED_ITEM` removes all quantities of the same item where as `DECREMENTED_ITEM_QUANTITY` removes only one per click